rc/lsp.kak

```diff
@@ -393,6 +393,68 @@ def lsp-diagnostic-lines-disable -docstring "Disable diagnostics line flags"  %{
     remove-highlighter global/lsp_error_lines
 }
 
+def lsp-find-error -params 0..2 -docstring "lsp-find-error [--pervious] [--include-warnings]
+Jump to the next or previous diagnostic error" %{
+    evaluate-commands %sh{
+        previous=false
+        errorCompare="DiagnosticError"
+        if [[ $1 == "--previous" ]]; then
```

Comment:
I have a feeling this isn't POSIX-compatible so won't work with `dash`, because https://github.com/mawww/kakoune/issues/2152 is being ignored using `dash` instead of `bash` is the only way to achieve reasonable startup performance.

Reply:
I used shellcheck to find non-posix compliant statements, so this should work in dash now too

Reply:
It still isn't POSIX-compliant.

Reply:
Do you have any specific points?

Reply:
In a POSIX shell, the test section of an if condition is not a builtin; instead, it uses `test(1)` in order to compute `true` or `false`.  In this case, you should do:

    [ "$1" = "--previous" ]

Reply:
> I used shellcheck to find non-posix compliant statements, so this should work in dash now too

@JJK96 I recommend `checkbashism` for this purpose. For example, for the line 

```sh
if [[ "$1" == "--previous" ]]; then
```

It outputs the following:

```
$ checkbashisms tmp.sh
possible bashism in tmp.sh line 5 (alternative test command ([[ foo ]] should be [ foo ])):
if [[ "$1" == "--previous" ]]; then
possible bashism in tmp.sh line 5 (should be 'b = a'):
if [[ "$1" == "--previous" ]]; then
```